### PR TITLE
[test] change boost random to std random in robustness tests

### PR DESCRIPTION
**Pull request by barendgehrels**

### CODE REVIEW

File: test/robustness/overlay/linear_areal/recursive_polygons_linear_areal.cpp [hunk 1]
```diff
@@ -202,22 +199,29 @@ public :
...
-        segment_type seg1, seg2;
-        bg::convert(m_segment, seg1);
-        bg::convert(segment, seg2);
-
-        // TODO: this function requires unique subranges now
-        bg::segment_intersection_points<pt> is = strategy.apply(seg1, seg2);
+        bg::detail::segment_as_subrange<segment_t> subrange1(seg1);
+        bg::detail::segment_as_subrange<segment_t> subrange2(seg2);
+        bg::segment_intersection_points<point_t> is 
+            = strategy.apply(subrange1, subrange2, policy_t());
```
Comment (barendgehrels) [inline test/robustness/overlay/linear_areal/recursive_polygons_linear_areal.cpp:224]: Additionally, it repairs this broken test.

File: test/robustness/overlay/buffer/multi_point_growth.cpp [hunk 2]
```diff
@@ -73,13 +73,12 @@ double test_growth(Geometry const& geometry, int n, int d, double distance)
     typedef bg::strategy::buffer::distance_symmetric<coordinate_type> distance_strategy_type;
     distance_strategy_type distance_strategy(distance);
 
-    std::vector<GeometryOut> buffered;
+    bg::model::multi_polygon<GeometryOut> buffered;
```
Comment (barendgehrels) [inline test/robustness/overlay/buffer/multi_point_growth.cpp:76]: This repairs this test.

File: test/robustness/convex_hull/random_multi_points.cpp [hunk 3]
```diff
@@ -9,30 +9,24 @@
...
+#include <chrono>
 #include <sstream>
 #include <fstream>
 
 #include <boost/program_options.hpp>
-#include <boost/random/linear_congruential.hpp>
-#include <boost/random/uniform_int.hpp>
-#include <boost/random/uniform_real.hpp>
-#include <boost/random/variate_generator.hpp>
-#include <boost/timer.hpp>
```
Comment (barendgehrels) [inline test/robustness/convex_hull/random_multi_points.cpp:17]: It also changes `boost::timer` to `std::chrono` in the touched files.

File: test/robustness/convex_hull/random_multi_points.cpp [hunk 4]
```diff
@@ -170,20 +165,21 @@ int main(int argc, char** argv)
...
-        int pcount = 3;
+        int seed = -1;
+        int pcount = 30;
         int field_size = 10;
         settings_type settings;
 
         description.add_options()
             ("help", "Help message")
             ("seed", po::value<int>(&seed), "Initialization seed for random generator")
-            ("count", po::value<int>(&count)->default_value(1), "Number of tests")
```
Comment (vissarion) [inline test/robustness/convex_hull/random_multi_points.cpp:181]: Is this not needed anymore? Seems unrelated to boost random replacement.

**Review by vissarion — APPROVED** [review]
Thanks! LGTM.

File: test/robustness/convex_hull/random_multi_points.cpp (hunk 4 above)
Comment (barendgehrels) [reply test/robustness/convex_hull/random_multi_points.cpp:181]: hmm, thanks! I have to check this indeed. Looks like an unwanted omission.

**Review by awulkiew — APPROVED** [review]
Thanks!

File: test/robustness/convex_hull/random_multi_points.cpp (hunk 4 above)
Comment (barendgehrels) [reply test/robustness/convex_hull/random_multi_points.cpp:181]: It's fine. Indeed not really unrelated to Boost.Random - but sidely because I changed some of these in touched code.
The `->default_value` is not necessary if the variable is already initialized like that. And in this case (and many more), it is.
